ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -26,103 +136,425 @@ describe('Dialog API', () => {
+            })
+          cy.withinDialog(() => {
+            closeDialogViaBackdrop()
+          })
+
+          cy.dataCy('input-field')
+            .should('have.focus')
+
+          // Test the input field not regaining focus
+          cy.dataCy('input-field')
+            .then(async () => {
+              await Cypress.vueWrapper.setProps({ noRefocus: true })
+              model.value = true
+              await nextTick()
+            })
+
+          cy.withinDialog(() => {
+            closeDialogViaBackdrop()
+          })
+          cy.dataCy('input-field')
+            .should('not.have.focus')
         })
       })
 
       describe('(prop): no-focus', () => {
-        it.skip(' ', () => {
-          //
+        it('should not focus on dialog when switching to it', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              noFocus: false
+            }
+          })
+
+          cy.withinDialog(() => {
+            cy.focused()
+              .should('have.class', 'q-dialog__inner')
+            closeDialogViaBackdrop()
+          })
+
+          cy.wrap().then(async () => {
+            await Cypress.vueWrapper.setProps({ noFocus: true })
+            model.value = true
+            await nextTick()
+          })
+
+          cy.withinDialog(() => {
+            cy.focused().should('not.exist')
+            closeDialogViaBackdrop()
+          })
         })
       })
 
       describe('(prop): no-shake', () => {
-        it.skip(' ', () => {
-          //
+        it('should not shake dialog', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              persistent: true
+            }
+          })
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaBackdrop()
+              cy.get('.q-dialog__inner')
+                .should('have.class', 'q-animate--scale')
+            }
+          })
+
+          assertPersistentDialogExists()
+
+          cy.wrap().then(async () => {
+            await Cypress.vueWrapper.setProps({ noShake: true })
+          })
+
+          cy.withinDialog({
+            persistent: true, fn: () => {
+              closeDialogViaBackdrop()
+              cy.get('.q-dialog__inner')
+                .should('not.have.class', 'q-animate--scale')
+            }
+          })
+
+          assertPersistentDialogExists()
         })
       })
     })
 
     describe('Category: content', () => {
       describe('(prop): seamless', () => {
-        it.skip(' ', () => {
-          //
+        it('should put the dialog in a seamless state', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              seamless: true
+            }
+          })
+
+          cy.dataCy('dialog-form')
```

Comment:
Could this be done using `withinDialog` and `cy.root()` within it?
Not sure, check it out

In general, it's not totally clear to me how we should deal with seamless dialogs

Reply:
I ended up using `cy.root()` within `withinDialog`